atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -267,6 +270,8 @@ private enum State {
     protected final SuccessCallbackManager successCallbackManager = new SuccessCallbackManager();
     private final CommitTimestampLoader commitTimestampLoader;
     private final ExpectationsMetrics expectationsDataCollectionMetrics;
+    private long cellCommitLocksRequested;
+    private long rowCommitLocksRequested;
```

Comment:
As the transaction may be accessed across multiple threads, we should consider making these AtomicInteger to avoid JMM word tearing as the long increments and reads/writes are not atomic or thread safe https://docs.oracle.com/javase/specs/jls/se8/html/jls-17.html#jls-17.7 or adjusting how we safely read/write these values

```suggestion
    private final AtomicInteger cellCommitLocksRequested = new AtomicInteger();
    private final AtomicInteger rowCommitLocksRequested = new AtomicInteger();
```

Reply:
Since we are only writing to the variable (and not incrementing/adding), volatile should be sufficient and more performant.

Reply:
@schlosna I wrote up a message but forgot to send, will summarise

For context:
The relevant variables are write at most once and read n times, and the write happens-before (explicitly) the reads to either happen, and so we care about visibility rather than atomicity here. (while it's possible to retrieve the commit lock information outside of the metric reporting API, that's explicitly a here-be-dragons request where we don't have any guarantees on the correctness of the data)

Thus, I'm pretty sure that volatile is sufficient here to provide said visibility, rather than requiring the stronger atomicity + visibility properties that AtomicInteger gives us.

That said, thanks for the review and the additional context on JMM word tearing!